packages/next/server/lib/start-server.ts

```diff
@@ -18,8 +19,21 @@ export default async function start(
   requestHandler = app.getRequestHandler()
 
   await new Promise<void>((resolve, reject) => {
-    // This code catches EADDRINUSE error if the port is already in use
-    srv.on('error', reject)
+    const isExplicitPort = serverOptions.inlinePort != null
+    srv.on('error', (err: NodeJS.ErrnoException) => {
+      // This code catches EADDRINUSE error if the port is already in use
+      if (
+        err.code === 'EADDRINUSE' &&
+        serverOptions.isNextDevCommand &&
+        !isExplicitPort
+      ) {
+        warn(`Port ${port} is in use, trying ${port! + 1} instead.`)
+        port! += 1
+        srv.listen(port, hostname)
+      } else {
+        reject(err)
+      }
+    })
```

Comment:
Take a look at `vercel dev` to see how it works. We should probably match the same behavior for `next dev` too 

https://github.com/vercel/vercel/blob/273718e0b78447bbd5b6092df720877099a18752/packages/cli/src/util/dev/server.ts#L845-L869

Reply:
Thanks for all the guidance.
I'm still new to next codebase, so I try to limit my changes to a minimum level. If we follow `vercel dev`, does it mean we should also make changes to these codes?

https://github.com/vercel/next.js/blob/5f19f766fdcad6efbfdb17777c0d7c364d408ba6/packages/next/cli/next-dev.ts#L102-L126